cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -43,25 +45,31 @@ def domain(request):
 
 @pytest.fixture(scope="module")
 def test_vm(setup_provider_modscope, provider, vm_name, request):
-    """Fixture to provision appliance to the provider being tested if necessary"""
-    vm = VM.factory(vm_name, provider, template_name=provider.data['full_template']['name'])
-
-    if not provider.mgmt.does_vm_exist(vm_name):
-        vm.create_on_provider(find_in_cfme=True, allow_skip="default")
-        request.addfinalizer(vm.delete_from_provider)
-    return vm
-
-
-def new_snapshot(test_vm, has_name=True):
+    def _test_vm(small_template=True):
+        """Fixture to provision appliance to the provider being tested if necessary"""
+        name = vm_name()
+        if small_template:
+            vm = VM.factory(name, provider, template_name=provider.data['small_template'])
+        else:
+            vm = VM.factory(name, provider, template_name=provider.data['full_template']['name'])
+
+        if not provider.mgmt.does_vm_exist(name):
+            vm.create_on_provider(find_in_cfme=True, allow_skip="default")
+            request.addfinalizer(vm.delete_from_provider)
+        return vm
+    return _test_vm
```

Comment:
**REQUIRED**
I would make two separate fixtures: `small_test_vm` and `full_test_vm`:
```python
def provision_vm(request, name, provider, small_template=True):
    if small_template:
        vm = VM.factory(name, provider, template_name=provider.data['small_template'])
    else: 
        vm = VM.factory(name, provider, template_name=provider.data['full_template']['name'])
    if not provider.mgmt.does_vm_exist(name):
        vm.create_on_provider(find_in_cfme=True, allow_skip="default")
        request.addfinalizer(vm.delete_from_provider)
    return vm

@pytest.fixture(scope="module")
def small_test_vm(request, setup_provider_modscope, provider, vm_name):
    return provision_vm(request, vm_name(), provider)

@pytest.fixture(scope="module")
def full_test_vm(request, setup_provider_modscope, provider, vm_name):
    return provision_vm(request, vm_name(), provider, small_template=False)
```